chain/chain/src/chain.rs

```diff
@@ -952,13 +952,17 @@ impl Chain {
         chain_store_update.commit()?;
 
         // clear all trie data
-        let mut store_update = StoreUpdate::new_with_tries(self.runtime_adapter.get_tries());
-        let stored_state = self.store().store().iter_prefix(ColState, &[]);
-        for (key, _) in stored_state {
+        let keys: Vec<Vec<u8>> =
+            self.store().store().iter_prefix(ColState, &[]).map(|kv| kv.0.into()).collect();
+        let tries = self.runtime_adapter.get_tries();
+        let mut chain_store_update = self.mut_store().store_update();
+        let mut store_update = StoreUpdate::new_with_tries(tries);
+        for key in keys.iter() {
             store_update.delete(ColState, key.as_ref());
+            chain_store_update.gc_col_state();
```

Comment:
let's move the whole logic of this function to the rest of gc and make stuff like `clear_chunk_data` private

Reply:
This looks default to move whole logic reset_data_pre_state_sync as a method of ChainStoreUpdate. reset_data_pre_state_sync does:
```
let mut chain_store_update = ...;
chain_store_update.commit();
...
let mut chain_store_update = ...;
chain_store_update.commit();
...
let mut chain_store_update = ...;
chain_store_update.commit();
...
```
But as a ChainStoreUpdate method it can only do one of above steps, to use mut ref self once